reactioncommerce/reaction
# Fix 4343 Cannot set replyTo or other field options when using Reaction.Email.send

Resolves #4343 Cannot set replyTo or other field options when using Reaction.Email.send
Impact: **minor**  
Type: **refactor**

## Issue
I am trying to modify a function responsible for sending emails to user by adding replyTo option.
After adding the replyTo option,the email is received by the user (so function keeps working), no errors appear in console but the replyTo option is not included in the fields the user sees in his email software.

## Solution
Solution is pretty straight forward as you will see.I've added *optionalEmailFields*  which will be destructured and thus added, if user decides that he wants more email options than the from, to, subject, html which are already set by default.

## Testing
1. Open a method used for sending emails for example *sendWelcomeEmail* method  on  server/methods/accounts/accounts.js
2. Add another nodemailer field like *replyTo* option in the Reaction.Email.send function like this:
 ```
Reaction.Email.send({
    to: userEmail,
    from: "someone@yyy.com",
    replyTo: "someoneElse@xxx.com", // ------- added here -------
    subject: SSR.render(subject, dataForEmail),
    html: SSR.render(tpl, dataForEmail)
  });
```
3. Register as a new user and you will see that in the email you'll receive you will have a distinct *reply to* field .

credit @artlimes
